### PR TITLE
[chore] Add issues with waiting-for-code-owners label to the weekly report

### PR DESCRIPTION
**Pull request by LucaLanziani**
**Description:** 

Added the list of issues with the `waiting-for-code-owner` to the weekly report

**Link to tracking Issue:** 

#32490

**Testing:**

Check the testing_actions.md file

**Documentation:**

I've added the docs/testing_actions.md file to describe how to test the new functionality

### CODE REVIEW

File: .github/workflows/scripts/generate-weekly-report.js
Comment (mx-psi) [inline .github/workflows/scripts/generate-weekly-report.js:159]:
```suggestion
      title: "New issues needing code owners",
```

File: .github/workflows/scripts/generate-weekly-report.js
Comment (mx-psi) [inline .github/workflows/scripts/generate-weekly-report.js:297]: Do we have an issue number for this? I think we should fix this first if possible instead of introducing a workaround, as I understand it's entirely under our control

File: .github/workflows/scripts/generate-weekly-report.js
Comment (mx-psi) [inline .github/workflows/scripts/generate-weekly-report.js:440]: Not opposed to this entirely, but I think this and the related discussion belongs on a separate PR.

Comment (LucaLanziani) [reply .github/workflows/scripts/generate-weekly-report.js:440]: Happy to move it to a different PR, this is helpful when adding new functionalities to the report imho

Comment (mx-psi) [reply .github/workflows/scripts/generate-weekly-report.js:440]: Please do! I agree that it is helpful but would prefer discussion on what tool is the best to use here and we can do that independently of this change

File: .github/workflows/scripts/generate-weekly-report.js
Comment (mx-psi) [reply .github/workflows/scripts/generate-weekly-report.js:297]: @crobert-1 any thoughts on how we should tackle this?

File: .github/workflows/scripts/generate-weekly-report.js
Comment (LucaLanziani) [reply .github/workflows/scripts/generate-weekly-report.js:440]: done

File: .github/workflows/scripts/generate-weekly-report.js
Comment (crobert-1) [reply .github/workflows/scripts/generate-weekly-report.js:297]: I'd prefer addressing separately, with a new issue opened for this. @LucaLanziani would you be able to open an issue?

Comment (LucaLanziani) [reply .github/workflows/scripts/generate-weekly-report.js:297]: Issue created https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33388

Comment (LucaLanziani) [reply .github/workflows/scripts/generate-weekly-report.js:297]: My suggestion would be to still merge this to fix the job while making the decision on issue #33388

The [JSON part](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/workflows/scripts/generate-weekly-report.js#L298-L305) (to be more precise the previous issue presence) was always optional and the code support that not being present, I would even "question" the initial idea of throwing the error given the optionality of the functionality.

Comment (LucaLanziani) [reply .github/workflows/scripts/generate-weekly-report.js:297]: oh sure, if the decision is already made in the issue (as I can see it is) happy to split the commits

File: .github/workflows/scripts/generate-weekly-report.js
Comment (crobert-1) [inline .github/workflows/scripts/generate-weekly-report.js:118]: This label is valid for both PRs and issues, so I don't think we should filter PRs out of the report.

For some context, there was discussion on this label in the January 31, 2024 sig meeting. The original intention was to label PRs that were stuck because they hadn't received approval from code owners yet. We've then repurposed it as well where an issue is open that needs input from code owners for a final decision.

File: .github/workflows/scripts/generate-weekly-report.js
Comment (crobert-1) [inline .github/workflows/scripts/generate-weekly-report.js:159]:
```suggestion
      title: "Issues and PRs that need code owner review",
```
The purpose of this label is essentially saying that the given issue or PR is essentially blocked until a code owner shares their thoughts. Whether it's an approval on a PR or "final say" on design decision in the component.

I'm open to different wording here, whatever is most clear.

**Comment by LucaLanziani** [comment]
Before: https://github.com/LucaLanziani/opentelemetry-collector-contrib/issues/3
After: https://github.com/LucaLanziani/opentelemetry-collector-contrib/issues/5

**Review by atoulme — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

**Review by crobert-1 — APPROVED** [review]
Thanks for adding this @LucaLanziani, really appreciate the help!

**Pull request merged**
